FHIR-SW-Development-Guide.ipynb

```diff
@@ -0,0 +1,1560 @@
+    "\n",
+    "### Conformance Resources\n",
+    "- An entity in FHIR is called a `Resource`. Examples: `Patient`, `Specimen`, `StructureDefinition`\n",
+    "- The phrase \"FHIR Model\" refers to a set of FHIR conformance resources\n",
+    "- Types of conformance resources include `StructureDefinition`, `CodeSystem`, `SearchParameter`.\n",
+    "- A `StructureDefinition` is essentially a schema. \n",
+    "- A `StructureDefinition` can represent the schema for a particular FHIR resource (e.g. Patient) or an attribute  on a FHIR resource (e.g. biological-sex)\n",
+    "- A `StructureDefinition` that represents an attribute on a FHIR resource (e.g. biological-sex) is called an Extension\n",
+    "- An Extension based `StructureDefinition` is used to add a new attribute to a FHIR resource\n",
+    "- A non-Extension based `StructureDefinition` is often called a `profile`\n",
+    "\n",
+    "### Terminology Resources\n",
+    "- A `CodeSystem` is a FHIR resource that represents an ontology\n",
+    "- A `CodeSystem` contains concepts and their codes \n",
+    "- A `ValueSet` is a restricted set of codes (e.g. `male` | `female`) from one or more `CodeSystems`\n",
+    "- A `ValueSet` can be bound to an attribute on a FHIR resource to restrict the values of the attribute\n",
+    "\n",
+    "\n",
+    "⚡️ **To \"develop a FHIR model\" means to extend the conformance resources in the base model**"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "heading_collapsed": true
+   },
+   "source": [
+    "## Objective\n",
+    "As a FHIR model developer, I've been tasked to extend the [NCPI Patient model](https://github.com/ncpi-fhir/ncpi-model-forge/blob/master/site_root/input/resources/profiles/StructureDefinition-ncpi-patient.json) with a new attribute to capture the Patient's species. These types of implementation tasks will be a result of the NCPI Project Forge modeling group's model discussions in which the FHIR model will be iteratively designed. \n",
+    "\n",
+    "As model developers, it is our job to implement the design in the form of FHIR conformance resources.\n",
+    "\n",
+    "I can find more details on exactly what should be implemented here in the [GitHub issue](https://github.com/ncpi-fhir/ncpi-model-forge/issues/6) that has been created for this task."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "heading_collapsed": true
+   },
+   "source": [
+    "## Create an Extension for Patient species 🖐"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "hidden": true
+   },
+   "source": [
+    "From reading the Github issue, I know I need to create an `Extension` type `StructureDefinition` resource to capture species of a Patient. \n",
+    "\n",
+    "There are a number of tools for authoring FHIR resources, however, most of the time it is easiest to search for existing similar extensions on the 📘 [HL7 FHIR website](https://www.hl7.org/fhir/extensibility-registry.html). \n",
```

Comment:
This registry website supports more accessible lookup: https://registry.fhir.org/